### PR TITLE
BAEL-2167 Added classes for article

### PR DESCRIPTION
**Pull request by priyeshmashelkar**

### CODE REVIEW

File: xml/src/test/java/com/baeldung/xml/XMLDocumentWriterUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import org.junit.Test;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+
+public class XMLDocumentWriterUnitTest {
+
+    @Test
+    public void givenXMLDocumentWhenWriteIsCalledThenXMLIsWrittenToFile() throws Exception {
+        Document document = createSampleDocument();
+        new XMLDocumentWriter().write(document, "company_simple.xml", false, false);
```
Comment (jzheaux) [inline xml/src/test/java/com/baeldung/xml/XMLDocumentWriterUnitTest.java]: My only concern here is that these tests will leave artifacts on the filesystem, right? Can we clean them up in an `@After` or similar?

File: xml/src/main/java/com/baeldung/xml/XMLDocumentWriter.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+            if(prettyPrint) {
+                transformer.setOutputProperty(OutputKeys.INDENT, "yes");
+                transformer.setOutputProperty("{http://xml.apache.org/xslt}indent-amount", "2");
+            }
+            DOMSource source = new DOMSource(document);
+            FileWriter writer = new FileWriter(new File(fileName));
+            StreamResult result = new StreamResult(writer);
+            transformer.transform(source, result);
+        } catch (IOException e) {
+            e.printStackTrace();
```
Comment (jzheaux) [inline xml/src/main/java/com/baeldung/xml/XMLDocumentWriter.java]: Let's not encourage this practice. At least, let's do `throw new IllegalStateException(e)` or `throw new IllegalArgumentException(e)`, depending on whether we think it is a system error or user error.

File: xml/src/test/java/com/baeldung/xml/XMLDocumentWriterUnitTest.java (hunk 1 above)
Comment (priyeshmashelkar) [reply xml/src/test/java/com/baeldung/xml/XMLDocumentWriterUnitTest.java]: Added cleanUp method

File: xml/src/main/java/com/baeldung/xml/XMLDocumentWriter.java (hunk 2 above)
Comment (priyeshmashelkar) [reply xml/src/main/java/com/baeldung/xml/XMLDocumentWriter.java]: Updated code as per suggestion

**Pull request merged**